### PR TITLE
feat(migrator): allow disabling logging via `migrations.silent`

### LINKED ISSUE #4124: Add option to make migrator 'quiet'
**Is your feature request related to a problem? Please describe.**
When I'm running tests via Jest I would like to drop schema before every test case, but that results in a lot of 'noise' in log output:
```
console.log
    Processing 'Migration20230222121600InitDb'

      at ../node_modules/@mikro-orm/migrations/Migrator.js:94:45
          at Array.map (<anonymous>)

console.log
    Applied 'Migration20230222121600InitDb'

      at ../node_modules/@mikro-orm/migrations/Migrator.js:95:44
          at Array.map (<anonymous>)
```

**Describe the solution you'd like**
Is it possible to add 'quiet' boolean option to migrator config? Something like this:
```
migrations: {
    quiet: true, // disables all log output from migrator, default: false
},
```

### PR DESCRIPTION
**Pull request by AnimeshRy**
This provides an option to make the migration logs silent.

Closes #4124

### CODE REVIEW

File: packages/migrations-mongodb/src/Migrator.ts [hunk 1]
```diff
@@ -84,10 +84,13 @@ export class Migrator implements IMigrator {
...
-    this.umzug.on('migrated', event => logger(`Applied '${event.name}'`));
-    this.umzug.on('reverting', event => logger(`Processing '${event.name}'`));
-    this.umzug.on('reverted', event => logger(`Reverted '${event.name}'`));
+
+    if (!(this.options?.silent)) {
+        this.umzug.on('migrating', event => logger(`Processing '${event.name}'`));
+        this.umzug.on('migrated', event => logger(`Applied '${event.name}'`));
+        this.umzug.on('reverting', event => logger(`Processing '${event.name}'`));
+        this.umzug.on('reverted', event => logger(`Reverted '${event.name}'`));
+      }
```
Comment (B4nan) [inline packages/migrations-mongodb/src/Migrator.ts:93]: bad indent

File: packages/migrations-mongodb/src/Migrator.ts [hunk 2]
```diff
@@ -84,10 +84,13 @@ export class Migrator implements IMigrator {
     });
 
     const logger = this.config.get('logger');
-    this.umzug.on('migrating', event => logger(`Processing '${event.name}'`));
-    this.umzug.on('migrated', event => logger(`Applied '${event.name}'`));
-    this.umzug.on('reverting', event => logger(`Processing '${event.name}'`));
-    this.umzug.on('reverted', event => logger(`Reverted '${event.name}'`));
+
+    if (!(this.options?.silent)) {
```
Comment (B4nan) [inline packages/migrations-mongodb/src/Migrator.ts:88]:
```suggestion
    if (!this.options.silent) {
```

File: packages/migrations/src/Migrator.ts [hunk 3]
```diff
@@ -116,10 +116,13 @@ export class Migrator implements IMigrator {
     });
 
     const logger = this.config.get('logger');
-    this.umzug.on('migrating', event => logger(`Processing '${event.name}'`));
-    this.umzug.on('migrated', event => logger(`Applied '${event.name}'`));
-    this.umzug.on('reverting', event => logger(`Processing '${event.name}'`));
-    this.umzug.on('reverted', event => logger(`Reverted '${event.name}'`));
+
+    if (!(this.options?.silent)) {
```
Comment (B4nan) [inline packages/migrations/src/Migrator.ts:120]: same here

File: packages/migrations-mongodb/src/Migrator.ts [hunk 4]
```diff
@@ -84,10 +84,13 @@ export class Migrator implements IMigrator {
     });
 
     const logger = this.config.get('logger');
```
Comment (B4nan) [inline packages/migrations-mongodb/src/Migrator.ts:86]: this line can also go inside the if block

**Review by B4nan — CHANGES_REQUESTED** [review]

File: packages/migrations/src/Migrator.ts (hunk 3 above)
Comment (AnimeshRy) [reply packages/migrations/src/Migrator.ts]: done

File: packages/migrations-mongodb/src/Migrator.ts (hunk 4 above)
Comment (AnimeshRy) [reply packages/migrations-mongodb/src/Migrator.ts]: done

**Comment by AnimeshRy** [comment]
@B4nan Can we enforce indent with eslint aswell ? I can't find the indent setting present in the current config file

**Comment by B4nan** [comment]
Thanks!

Feel free to send a separate PR improving the eslint config 👍

**Pull request merged**